### PR TITLE
Jdt replace regexp with plain replace clean up

### PR DESCRIPTION
**Pull request by cal101**

### CODE REVIEW

File: org.eclipse.m2e.core.ui/src/org/eclipse/m2e/core/ui/internal/wizards/MavenProjectWizardArchetypePage.java [hunk 1]
```diff
@@ -438,7 +438,7 @@ public void inputChanged(Viewer viewer, Object oldInput, Object newInput) {
         String description = archetype.getDescription();
 
         String text = description == null ? "" : description; //$NON-NLS-1$
-        text = text.replaceAll("\n", "").replaceAll("\\s{2,}", " "); //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$ //$NON-NLS-4$
+        text = text.replace("\n", "").replaceAll("\\s{2,}", " "); //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$ //$NON-NLS-4$
```
Comment (fbricon) [inline org.eclipse.m2e.core.ui/src/org/eclipse/m2e/core/ui/internal/wizards/MavenProjectWizardArchetypePage.java:441]: seems wrong

Comment (fbricon) [reply org.eclipse.m2e.core.ui/src/org/eclipse/m2e/core/ui/internal/wizards/MavenProjectWizardArchetypePage.java:441]: my bad, it's fine

Comment (cal101) [reply org.eclipse.m2e.core.ui/src/org/eclipse/m2e/core/ui/internal/wizards/MavenProjectWizardArchetypePage.java:441]: Exactly the phases I passed myself when first seeing that ...

**Comment by mickaelistria** [comment]
Those commits need to be rebased to workaround the version issue happening on CI (we're in one of the rare case when this issue may happen because of numerous commits merged in some random-ish order). Can you please rebase and then `git push --force` to your fork branch?

**Comment by mickaelistria** [comment]
Looks like this change needs a rebase (this happens when changes are happening in a short time frame and merged in a different order than the one they were authored).

**Comment by cal101** [comment]
@mickaelistria @fbricon @ifedorenko I don't understand why this change makes a second test fail. So I don't have a clue what to change or do differently. It would be very nice if one of the m2e experts can help to explain.

**Pull request merged**